package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py

```diff
@@ -316,9 +316,17 @@ def __init__(self, universe,
 
         self.u = universe
         self._trajectory = self.u.trajectory
-        self.donors_sel = donors_sel
-        self.hydrogens_sel = hydrogens_sel
-        self.acceptors_sel = acceptors_sel
+
+        msg = ("{} is an empty selection string - no hydrogen bonds will "
+               "be found. This may be intended, but please check your "
+               "selection."
+        )
+        if isinstance(donors_sel, str) and not donors_sel:
```

Comment:
I feel like this could be squashed into a `for` loop, thoughts?

Reply:
I've put those lines into a for loop now. It's slightly less readable to me, but that might just be me, and I'm happy to leave it this way